### ISSUE TITLE
Cross reference variables inside the configuration file

### ISSUE THREAD

**Issue by kboom** [sksamuel/hoplite#116]
It seems reasonable to me that the following should be possible

```yaml
apigee:
  identityUrl: https://something.com/else

security:
  host: ${apigee.identityUrl}
```

However it is not. Would it be possible to implement this requirement?

**Comment by sksamuel** [comment]
Should be easy to do. Do you think it should run across the merged config
tree or just inside the single file ?

**Comment by sksamuel** [comment]
Included in 1.3.0
